File: Leetcode/easy/isSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        Integer leftCounter = 0;

        for (int rightCounter = 0; rightCounter < t.length(); rightCounter++) {
            try {
                if (t.charAt(rightCounter) == s.charAt(leftCounter)) {
                    leftCounter++;
                }
            } catch (Exception e) {
                continue;
            }
            
        }

        if (leftCounter != s.length()) {
            return false;
        }

        return true;
    }
}

/*

Runtime
2 ms
Beats
70.58%
Memory
40.6 MB
Beats
81.45%

 */

File: Leetcode/easy/isSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        count = 0

        for num in range(len(t)):
            try:
                if s[count] == t[num]:
                    count += 1
            except IndexError:
                continue

        if count == len(s):
            return True
        else:
            return False

'''

Runtime
33 ms
Beats
92.61%
Memory
13.9 MB
Beats
82.15%

'''
